File: challenges/151/python/solution.py
#!/usr/bin/env python
from challenges.utils import expect


def reverse_words(s: str) -> str:
    if not s:
        return s

    return " ".join([word for word in reversed(s.split(" ")) if word])


if __name__ == "__main__":
    expect(reverse_words("  hello world  "), "world hello")
    expect(reverse_words("a good   example"), "example good a")

File: challenges/151/js/solution.ts
import { expect } from "../../util";

const reverseWords = (s: string): string => {
  return s
    .split(" ")
    .filter((word) => word)
    .reverse()
    .join(" ");
};

expect(reverseWords("  hello world  "), "world hello");
expect(reverseWords("a good   example"), "example good a");
